packages/core/src/notification/NotificationService.ts

```diff
@@ -231,42 +229,25 @@ export class NotificationService extends TypedEventEmitter<Events> {
   /**
    * Will process one event
    * @param event The backend event to process
-   * @param source The source of the event (websocket or notication stream)
    * @param dryRun Will not try to decrypt if true
    * @return the decrypted payload and the raw event. Returns `undefined` when the payload is a coreCrypto-only system message
    */
-  private async handleEvent(
-    event: BackendEvent,
-    source: NotificationSource,
-    dryRun: boolean = false,
-  ): Promise<HandledEventPayload | undefined> {
-    // Handle MLS Events
-    const mlsResult = await this.mlsService?.handleEvent({event, source, dryRun});
-    if (mlsResult) {
-      return mlsResult;
-    }
-
-    const proteusResult = await this.proteusService.handleEvent({
-      event,
-      source,
-      dryRun,
-    });
-    if (proteusResult) {
-      return proteusResult;
+  private async handleEvent(event: BackendEvent, dryRun: boolean = false): Promise<HandledEventPayload | undefined> {
+    if (dryRun) {
+      // In case of a dry run, we do not want to decrypt messages
+      // We just return the raw event to the caller
+      return {event};
     }
 
-    // Fallback to other events
     switch (event.type) {
-      // Meta events
-      case CONVERSATION_EVENT.MEMBER_JOIN:
-        // As of today (07/07/2022) the backend sends `WELCOME` message to the user's own conversation (not the actual conversation that the welcome should be part of)
-        // So in order to map conversation Ids and groupId together, we need to first fetch the conversation and get the groupId linked to it.
-        const conversation = await this.apiClient.api.conversation.getConversation(
-          event.qualified_conversation ?? {id: event.conversation, domain: ''},
-        );
-        if (!conversation) {
-          throw new Error('no conv');
-        }
+      case CONVERSATION_EVENT.MLS_MESSAGE_ADD:
+        return this.conversationService.handleMLSMessageAddEvent(event);
+
+      case CONVERSATION_EVENT.MLS_WELCOME_MESSAGE:
+        return this.conversationService.handleMLSWelcomeMessageEvent(event);
+
+      case CONVERSATION_EVENT.OTR_MESSAGE_ADD:
+        return this.conversationService.handleOtrMessageAddEvent(event);
```

Comment:
It makes more sense to me to check the event type first and then handle the particular event type, than run a function responsible for handling it without even knowing the type yet.

Reply:
hummm actually, I'd think the other way around, the `NotificationService` should not be aware of the different types of events that come it. It should be as dumb as possible and the different handlers should be the one responsible for switching accros the different cases. 

I'd move this logic to the `conversationService` and make the `handle*` methods private (so that it doesn't leak the implementation). 

WDYT?